### ISSUE TITLE
Getting this error in v78

### ISSUE THREAD

**Issue by nick-77** [Tobi823/ffupdater#431]
Stacktrace:
```
java.lang.ClassCastException
	at androidx.appcompat.app.ToolbarActionBar$$ExternalSyntheticThrowCCEIfNotNull0.m(SourceFile:0)
	at de.marmaro.krt.ffupdater.network.fdroid.CustomRepositoryConsumer.getLatestUpdate(Unknown Source:112)
	at de.marmaro.krt.ffupdater.network.fdroid.CustomRepositoryConsumer$getLatestUpdate$1.invokeSuspend(Unknown Source:16)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(Unknown Source:11)
	at kotlinx.coroutines.DispatchedTask.run(Unknown Source:88)
	at android.os.Handler.handleCallback(Handler.java:942)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:240)
	at android.os.Looper.loop(Looper.java:351)
	at android.app.ActivityThread.main(ActivityThread.java:8399)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:584)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1013)
	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@aedf11e, Dispatchers.Main]
Timestamp: 2023-07-05T01:04:19.261

```
Device information:
| Key | Value |
| --- | --- |
| FFUpdater version | 78.0.0 (142) release |
| Device | RMX3301 (RMX3301, RED8ACL1, taro) |
| Manufacturer | realme (realme) | 
| Supported ABIs | arm64-v8a, armeabi-v7a, armeabi |
| Android version | 13 (SDK: 33) |
| OS | dg02-pool03-kvm95, root, release-keys, 1686542704000 |

**Comment by 1alessandro1** [comment]
Same
Stacktrace:
```
java.lang.ClassCastException
	at androidx.appcompat.app.ToolbarActionBar$$ExternalSyntheticThrowCCEIfNotNull0.m(SourceFile:0)
	at de.marmaro.krt.ffupdater.network.fdroid.CustomRepositoryConsumer.getLatestUpdate(Unknown Source:112)
	at de.marmaro.krt.ffupdater.network.fdroid.CustomRepositoryConsumer$getLatestUpdate$1.invokeSuspend(Unknown Source:16)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(Unknown Source:11)
	at kotlinx.coroutines.DispatchedTask.run(Unknown Source:88)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7833)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@904cfa2, Dispatchers.Main]
Timestamp: 2023-07-04T22:51:34.772

```
Device information:
| Key | Value |
| --- | --- |
| FFUpdater version | 78.0.0 (142) release |
| Device | M2012K11AG (lineage_alioth, alioth, kona) |
| Manufacturer | POCO (Xiaomi) | 
| Supported ABIs | arm64-v8a, armeabi-v7a, armeabi |
| Android version | 12 (SDK: 32) |
| OS | android-build, nobody, release-keys, 1679597388000 |

**Comment by Kurty00** [comment]
Same here thing here.
Stacktrace:
```
java.lang.ClassCastException
	at androidx.appcompat.app.ToolbarActionBar$$ExternalSyntheticThrowCCEIfNotNull0.m(SourceFile:0)
	at de.marmaro.krt.ffupdater.app.impl.Chromium.findStorageObject(Unknown Source:119)
	at de.marmaro.krt.ffupdater.app.impl.Chromium.access$findStorageObject(Unknown Source:0)
	at de.marmaro.krt.ffupdater.app.impl.Chromium$findStorageObject$1.invokeSuspend(Unknown Source:12)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(Unknown Source:11)
	at kotlinx.coroutines.DispatchedTask.run(Unknown Source:88)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7894)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1009)
	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@5c30a87, Dispatchers.Main]
Timestamp: 2023-07-04T22:57:52.227

```
Device information:
| Key | Value |
| --- | --- |
| FFUpdater version | 78.0.0 (142) release |
| Device | FP4 (FP4eea, FP4, FP4) |
| Manufacturer | Fairphone (Fairphone) | 
| Supported ABIs | arm64-v8a, armeabi-v7a, armeabi |
| Android version | 12 (SDK: 31) |
| OS | T2M-NB-SW-T1, android-new, release-keys, 1683710541000 |

**Comment by Tobi823** [comment]
How can I reproduce the error?

**Comment by Tobi823** [comment]
typical ... it runs on the emulator but not on my phone

**Comment by Tobi823** [comment]
Proguard breaks the release version - I guess that classes are removed which are later needed.

**Comment by Tobi823** [comment]
Fixed in 78.0.1. If not, please create a new issue.

**Comment by nick-77** [comment]
Thankyou sir
